src/server/config.ts

```diff
@@ -253,11 +260,11 @@ const CLUSTER_TYPES: SupportedType[] = ['druid', 'postgres', 'mysql'];
 
 var settingsLocation: SettingsLocation = null;
 if (serverSettingsFilePath) {
-  settingsLocation = {
-    location: 'local',
-    readOnly: false, // ToDo: this should be true
-    uri: serverSettingsFilePath
-  };
+  if (SERVER_SETTINGS.settingsUri) {
```

Comment:
I'm thinking maybe a `writable` flag on the AppSettings could be useful ? To put the UI in `read-only` mode or something ? Not necessarily for now though